webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpParallelCoordinates.tsx

```diff
@@ -97,47 +109,56 @@ const HpParallelCoordinates: React.FC<Props> = ({
     return undefined;
   }, [experiment.config.searcher, selectedMetric]);
 
-  const handleFilterChange = useCallback(
-    (hermesFilters: Hermes.Filters) => {
-      // Skip if there aren't any chart data.
-      if (!chartData) return;
+  const resetFilteredTrials = () => {
+    // Skip if there aren't any chart data.
+    if (!chartData) return;
 
-      // Initialize a new trial id filter map.
-      const newFilteredTrialIdMap = chartData.trialIds.reduce((acc, trialId) => {
-        acc[trialId] = true;
-        return acc;
-      }, {} as Record<number, boolean>);
+    // Initialize a new trial id filter map.
+    const newFilteredTrialIdMap = chartData.trialIds.reduce((acc, trialId) => {
+      acc[trialId] = true;
+      return acc;
+    }, {} as Record<number, boolean>);
 
-      // Figure out which trials are filtered out based on user filters.
-      Object.entries(hermesFilters).forEach(([key, list]) => {
-        if (!chartData.data[key] || list.length === 0) return;
+    // Figure out which trials are filtered out based on user filters.
 
-        chartData.data[key].forEach((value, index) => {
-          let isWithinFilter = false;
+    Object.entries(hermesCreatedFeatures).forEach(([key, list]) => {
+      if (!chartData.data[key] || list.length === 0) return;
 
-          list.forEach((filter: Hermes.Filter) => {
-            if (value >= filter[0] && value <= filter[1]) isWithinFilter = true;
-          });
+      chartData.data[key].forEach((value, index) => {
+        let isWithinFilter = false;
 
-          if (!isWithinFilter) {
-            const trialId = chartData.trialIds[index];
-            newFilteredTrialIdMap[trialId] = false;
+        list.forEach((filter: HermesInternalFilter) => {
+          const min = Math.min(Number(filter.value0), Number(filter.value1));
+          const max = Math.max(Number(filter.value0), Number(filter.value1));
+          if (value >= min && value <= max) {
+            isWithinFilter = true;
           }
         });
+
+        if (!isWithinFilter) {
+          const trialId = chartData.trialIds[index];
+          newFilteredTrialIdMap[trialId] = false;
+        }
       });
+    });
 
-      setFilteredTrialIdMap(newFilteredTrialIdMap);
-    },
-    [chartData],
-  );
+    setFilteredTrialIdMap(newFilteredTrialIdMap);
+  };
+
+  useEffect(() => {
+    resetFilteredTrials();
+  }, [chartData, hermesCreatedFeatures, resetFilteredTrials]);
```

Comment:
I'm getting "update depth exceeded" on this branch, I think because `resetFilteredTrials` is reassigned on each render.

changing this dependency array to `[resetFilteredTrials]` and wrapping `resetFiltered` trials in a 

```
useCallback(() => {
    ...
}, [chartData, hermesCreatedFeatures]);
```

resolved for me.

Reply:
just so I understand this right, it would be

```typescript
const resetFilteredTrials = useCallback(() => {
  // ...existing resetFilteredTrials code...
}, [chartData, hermesCreatedFeatures]);

useEffect(() => {
  resetFilteredTrials();
}, [resetFilteredTrials]);
```

Reply:
worked for me anyways :D